packages/lodestar-fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -0,0 +1,771 @@
+/* eslint-disable max-len */
+import {fromHexString, toHexString, readOnlyForEach, readOnlyMap} from "@chainsafe/ssz";
+import {
+  Slot,
+  ValidatorIndex,
+  BeaconBlock,
+  Root,
+  BeaconState,
+  IndexedAttestation,
+  Gwei,
+  Checkpoint,
+} from "@chainsafe/lodestar-types";
+import {
+  computeSlotsSinceEpochStart,
+  computeStartSlotAtEpoch,
+  computeEpochAtSlot,
+  ZERO_HASH,
+} from "@chainsafe/lodestar-beacon-state-transition";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+
+import {computeDeltas, HEX_ZERO_HASH, IVoteTracker, ProtoArray} from "../protoArray";
+import {ForkChoiceError, ForkChoiceErrorCode, InvalidBlockCode, InvalidAttestationCode} from "./errors";
+import {IForkChoiceStore} from "./store";
+import {IBlockSummary, toBlockSummary} from "./blockSummary";
+import {ILatestMessage, IQueuedAttestation} from "./interface";
+
+/**
+ * Provides an implementation of "Ethereum 2.0 Phase 0 -- Beacon Chain Fork Choice":
+ *
+ * https://github.com/ethereum/eth2.0-specs/blob/v0.12.2/specs/phase0/fork-choice.md#ethereum-20-phase-0----beacon-chain-fork-choice
+ *
+ * ## Detail
+ *
+ * This class wraps `ProtoArray` and provides:
+ *
+ * - Management of validators latest messages and balances
+ * - Management of the justified/finalized checkpoints as seen by fork choice
+ * - Queuing of attestations from the current slot
+ *
+ * This class MUST be used with the following considerations:
+ *
+ * - Time is not updated automatically, updateTime MUST be called every slot
+ * - Justified balances are not updated automatically, updateBalances MUST be called when Store justifiedCheckpoint is updated
+ */
+export class ForkChoice {
```

Comment:
Why not create IForkChoice interface and rename this class to ProtoArrayForkChoice?

I see multiple pros for that:
- when we initialize forkchoice in code we know exactly what implementation we are using
- we can add different implementations (faster, slower etc) without breaking api
- it's easier to stub it in tests

 If at some point we wan't to experiment with different implementations